README.md
#APESuperHUD
A simple way to display a HUD with a message or progress information in your application.

<p align="center">
  <img src="https://cloud.githubusercontent.com/assets/16682908/12681053/f4974f7e-c6ac-11e5-972e-f234200872aa.gif" alt="APESuperHUD">
  <img src="https://cloud.githubusercontent.com/assets/16682908/12681280/1c018a6a-c6ae-11e5-884e-d12a1a112f58.gif" alt="APESuperHUD">
</p>

##Features
- Simplicity.
- Customizable. See [Change appearance](#change-appearance).
- Fully written in Swift.
- Unit tests.

##Requirements
- iOS 8 or later.
- Xcode 7 or later.

##Installation
####CocoaPods
You can use [CocoaPods](http://cocoapods.org/) to install `APESuperHUD` by adding it to your `Podfile`:
```ruby
platform :ios, '8.0'
use_frameworks!

target 'MyApp' do
    pod 'APESuperHUD', :git => 'https://github.com/apegroup/APESuperHUD.git'
end
```
Note that this requires CocoaPods version 36, and your iOS deployment target to be at least 8.0:

####Carthage
You can use [Carthage](https://github.com/Carthage/Carthage) to install `APESuperHUD` by adding it to your `Cartfile`:
```
github "apegroup/APESuperHUD"
```

##Usage
Don't forget to import.
```swift
import APESuperHUD
```
####Show message HUD
######With default icon
```swift
APESuperHUD.showOrUpdateHUD(.Email, message: "1 new message", duration: 3.0, presentingView: self.view, completion: { _ in
    // Completed
})
```
######With custom image
```swift
APESuperHUD.showOrUpdateHUD(UIImage(named: "apegroup")!, message: "Demo message", duration: 3.0, presentingView: self.view, completion: { _ in
    // Completed
})
```

####Show HUD with loading indicator
######With loading text
```swift
APESuperHUD.showOrUpdateHUD(.Standard, message: "Demo loading...", presentingView: self.view, completion: nil)
```
######Without loading text
```swift
APESuperHUD.showOrUpdateHUD(.Standard, message: "", presentingView: self.view, completion: nil)
```
######With funny loading texts
```swift
APESuperHUD.showOrUpdateHUD(.Standard, funnyMessagesLanguage: .English, presentingView: self.view, completion: nil)
```
####Remove HUD
```swift
APESuperHUD.removeHUD(animated: true, presentingView: self.view, completion: { _ in
    // Completed
})
```
####Change appearance
```swift
APESuperHUD.appearance.cornerRadius = 12
APESuperHUD.appearance.animateInTime = 1.0
APESuperHUD.appearance.animateOutTime = 1.0
APESuperHUD.appearance.backgroundBlurEffect = .None
APESuperHUD.appearance.iconColor = UIColor.greenColor()
APESuperHUD.appearance.textColor = UIColor.greenColor()
APESuperHUD.appearance.loadingActivityIndicatorColor = UIColor.greenColor()
APESuperHUD.appearance.defaultDurationTime = 4.0
APESuperHUD.appearance.cancelableOnTouch = true
APESuperHUD.appearance.iconWidth = 48
APESuperHUD.appearance.iconHeight = 48
APESuperHUD.appearance.fontName = "Caviar Dreams"
APESuperHUD.appearance.fontSize = 14
```

##Contributing
All people are welcome to contribute. See CONTRIBUTING for details.

##License
APESuperHUD is released under the MIT license. See LICENSE for details.


CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

---

## [0.5.4](https://github.com/apegroup/APESuperHUD/releases/tag/0.5.4)

Centre the icon and the loading indicator if there is no message to show.

## [0.5.3](https://github.com/apegroup/APESuperHUD/releases/tag/0.5.3)

Prepare for SPM. Current version of SPM do not support UIKit.

On Dec 4, 2015, at 5:39 PM, Daniel Dunbar (@apple.com) wrote:
Right, now we only compile for the host platform (OS X or Linux, currently). Among other things, we currently have no knowledge (or options to choose) what SDK or architecture you are targeting. We also have no mechanisms for specifying what platforms targets are compatible with in the manifest.

## [0.5.2](https://github.com/apegroup/APESuperHUD/releases/tag/0.5.2)

Adds documentation for public enums and properties.

## [0.5.1](https://github.com/apegroup/APESuperHUD/releases/tag/0.5.1)

Faster display the HUD.
Remove deprecations introduced in Swift 2.2.

## [0.4.1](https://github.com/apegroup/APESuperHUD/releases/tag/0.4.1)

Fixes unit tests.

## [0.4](https://github.com/apegroup/APESuperHUD/releases/tag/0.4)

Updates demo project due to API modifications.

## [0.3.3](https://github.com/apegroup/APESuperHUD/releases/tag/0.3.3)

Deployment Target update.

## [0.3.2](https://github.com/apegroup/APESuperHUD/releases/tag/0.3.2)

Bugfix cocoapods.

## [0.3.1](https://github.com/apegroup/APESuperHUD/releases/tag/0.3.1)

Bugfix cocoapods.

## [0.3](https://github.com/apegroup/APESuperHUD/releases/tag/0.3)

Adds the possibility to specify the HUD size in HUDAppearance.

## [0.2](https://github.com/apegroup/APESuperHUD/releases/tag/0.2)

Test version.

## [0.1](https://github.com/apegroup/APESuperHUD/releases/tag/0.1)

Test version.

Demo/APESuperHUDDemo/ViewController.swift
// ViewController.swift
//
// The MIT License (MIT)
//
// Copyright (c) 2016 apegroup
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.

import UIKit
import APESuperHUD

class ViewController: UIViewController {
    
    @IBOutlet weak var messageWithDefaultIconButton: UIButton!
    @IBOutlet weak var messageWithCustomIconButton: UIButton!
    @IBOutlet weak var loadingWithTextButton: UIButton!
    @IBOutlet weak var loadingWithoutTextButton: UIButton!
    @IBOutlet weak var loadingWithFunnyMessagesButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Setup Appearnce
        setupHudAppearance()
        
        // Setup Buttons
        setupButtonsLayout()
    }
    
    private func setupHudAppearance() {
        APESuperHUD.appearance.cornerRadius = 12
        APESuperHUD.appearance.animateInTime = 1.0
        APESuperHUD.appearance.animateOutTime = 1.0
        APESuperHUD.appearance.backgroundBlurEffect = .None
        APESuperHUD.appearance.iconColor = UIColor(red: 38/255, green: 38/255, blue: 38/255, alpha: 1)
        APESuperHUD.appearance.textColor = UIColor(red: 38/255, green: 38/255, blue: 38/255, alpha: 1)
        APESuperHUD.appearance.loadingActivityIndicatorColor = UIColor(red: 38/255, green: 38/255, blue: 38/255, alpha: 1)
        APESuperHUD.appearance.defaultDurationTime = 4.0
        APESuperHUD.appearance.cancelableOnTouch = true
        APESuperHUD.appearance.iconWidth = 48
        APESuperHUD.appearance.iconHeight = 48
        APESuperHUD.appearance.fontName = "Caviar Dreams"
        APESuperHUD.appearance.fontSize = 14
    }
    
    private func setupButtonsLayout() {
        loadingWithTextButton.layer.cornerRadius = loadingWithFunnyMessagesButton.frame.height / 2
        loadingWithoutTextButton.layer.cornerRadius = loadingWithoutTextButton.frame.height / 2
        loadingWithFunnyMessagesButton.layer.cornerRadius = loadingWithFunnyMessagesButton.frame.height / 2
        messageWithDefaultIconButton.layer.cornerRadius = messageWithDefaultIconButton.frame.height / 2
        messageWithCustomIconButton.layer.cornerRadius = messageWithCustomIconButton.frame.height / 2
    }
    
    @IBAction func withDefaultIconButtonPressed(sender: UIButton) {
        APESuperHUD.showOrUpdateHUD(icon: .Email, message: "1 new message", duration: 3.0, presentingView: self.view, completion: { _ in
            // Completed
        })
    }
    
    @IBAction func withCustomIconButtonPressed(sender: UIButton) {
        APESuperHUD.showOrUpdateHUD(icon: UIImage(named: "apegroup")!, message: "Demo message", duration: 3.0, presentingView: self.view, completion: { _ in
            // Completed
        })
    }
    
    @IBAction func withLoadingTextButtonPressed(sender: UIButton) {
        APESuperHUD.showOrUpdateHUD(loadingIndicator: .Standard, message: "Demo loading...", presentingView: self.view)
        
        runWithDelay(3.0, closure: { [weak self] in
            APESuperHUD.showOrUpdateHUD(icon: .CheckMark, message: "Done loading!", duration: 2.0, presentingView: self!.view, completion: nil)
        })
    }
    
    @IBAction func withoutLoadingTextButtonPressed(sender: UIButton) {
        APESuperHUD.showOrUpdateHUD(loadingIndicator: .Standard, message: "", presentingView: self.view)
        
        runWithDelay(3.0, closure: { [weak self] in
            APESuperHUD.removeHUD(animated: true, presentingView: self!.view, completion: nil)
        })
    }
    
    @IBAction func withFunnyLoadingTextButtonPressed(sender: UIButton) {
        APESuperHUD.showOrUpdateHUD(loadingIndicator: .Standard, funnyMessagesLanguage: .English, presentingView: self.view)
        
        runWithDelay(10.0, closure: { [weak self] in
            APESuperHUD.removeHUD(animated: true, presentingView: self!.view, completion: nil)
        })
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

extension ViewController {
    
    func runWithDelay(delay: Double, closure: Void -> Void) {
        let time = dispatch_time(DISPATCH_TIME_NOW, Int64(delay * Double(NSEC_PER_SEC)))
        dispatch_after(time, dispatch_get_main_queue(), closure)
    }
}

